src/main/java/chatbot/logic/Parser.java

```diff
@@ -0,0 +1,172 @@
+package chatbot.logic;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import chatbot.exception.DeadlineArgsException;
+import chatbot.exception.EmptyArgsException;
+import chatbot.exception.EventArgsException;
+import chatbot.exception.InputException;
+import chatbot.exception.InvalidCommandException;
+import chatbot.exception.MissingDeadlineByException;
+import chatbot.exception.MissingEventFromException;
+import chatbot.exception.MissingEventToException;
+import chatbot.task.Deadline;
+import chatbot.task.Event;
+import chatbot.task.Task;
+import chatbot.task.Todo;
+
+/**
+ * Represents the parser, converting strings into chatbot instructions
+ * Either processes the user's inputs, or processes file inputs into instructions for the chatbot
+ */
+public class Parser {
+    /**
+     * Reads the encoded strings present in the data file, and processes it into a suitable Task
+     *
+     * @param taskString The encode string read from the file
+     * @return A Task object that represents the encoded string
+     * @throws IOException Exception thrown if string format is unknown
+     */
+    public static Task parseFileLine(String taskString) throws IOException {
+        // TODO error handling for if data file is corrupted
+        String[] tokens = taskString.split("\\|");
+        boolean isDone = Integer.parseInt(tokens[1]) == 1;
+        String title = tokens[2];
+        return switch (tokens[0]) {
+        case "T" -> new Todo(title, isDone);
```

Comment:
i believe there should be 4(?) spaces of indentation for every line after `switch`